js/main.js

```diff
@@ -304,6 +304,12 @@ function check_load() {
            console.log("changing download link to:", downloadConfigURL+"?sess_id="+sess_id)
            $("#download_config").attr("href", downloadConfigURL+"?sess_id="+sess_id);
        }
+      //  check if session cookie exists, if not, set it manually
+       
+        if (getCookie("sessionid") == null || getCookie("sessionid") == "") {
+          console.log("* setting session cookie: ", global_session_id)
+          document.cookie = "sessionid="+global_session_id;
```

Comment:
What's the issue with having the server send up a session id?

Reply:
cross domain cookies

Reply:
I'm going to ned a little more context. This looks like many clients will be using the same session ID to me.

Reply:
global_session_id is a variable for the session_key sent back to the user (it's not a static id), if you look a few lines up it's just storing the session id that the server gives. It's a solution to the cookie problem, this might be a smarter way to do it regardless